keps/sig-storage/transfer-pvc-between-namespaces.md

```diff
@@ -0,0 +1,176 @@
+---
+title: Add ability to transfer PVC between Namespaces
+authors:
+  - "@j-griffith"
+owning-sig: sig-storage
+participating-sigs:
+  - sig-architecture
+reviewers:
+  - TBD
+approvers:
+  - @saad-ali
+  - @childsb
+editor: @j-griffith
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: provisional
+---
+
+# Kubernetes PVC Namespace Transfer
+
+
+## Table of Contents
+
+* [Table of Contents](#table-of-contents)
+* [Summary](#summary)
+* [Motivation](#motivation)
+    * [Goals](#goals)
+    * [Non-Goals](#non-goals)
+* [Proposal](#proposal)
+    * [User Stories [optional]](#user-stories-optional)
+      * [Story 1](#story-1)
+      * [Story 2](#story-2)
+    * [Implementation Details/Notes/Constraints [optional]](#implementation-detailsnotesconstraints-optional)
+    * [Risks and Mitigations](#risks-and-mitigations)
+* [Graduation Criteria](#graduation-criteria)
+* [Implementation History](#implementation-history)
+* [Drawbacks [optional]](#drawbacks-optional)
+* [Alternatives [optional]](#alternatives-optional)
+
+[Tools for generating]: https://github.com/ekalinin/github-markdown-toc
+
+## Summary
+
+Provide a mechanism to allow users to "give" a PVC to a user in another namespace based on mutual acceptance. A transfer should be initiated by updating an existing PVC with a transfer request annotation, and accepted by the intended destination namespace by the recipient requesting a PVC using the standard mechanism and specifying the correct PVC Name and transfer annotation as per the transfer request.
+
+* Provide a Cluster wide mechanism to enable or disable the transfer of PVCs
+* Expose standardized process for transferring a PVC from one Namespace to another
+* Require explicit acceptance of a PVC into another users Namespace, ensuring a malitious user can't just dump PVCs into another users Namespace
+* Utilize existing PVC creation semantics for creating a PVC from a Transfer
+* Honor existing configuration options including quota checks, Storage Class access etc
+
+The process of transferring a PVC/Volume is as follows:
+1. Original user indicates they're willing to ``give`` the volume to another namespace
+2. The receiving user indicates they'd like to ``accept`` the volume into their namespace
+3. When the original user deletes the PVC from their namespace, the claim request from the receiving user is fulfilled
+
+## Motivation
+
+There are a number of uses cases where a user would like to have the ability to transfer an existing PVC from one namespace to another.  This is a valuable workflow for persistent storage and enables the ability to easily duplicate and transfer data sets from one environment to another.  These populated PVCs could be a clone of another volume, a volume from snapshot, or data that was written to the volume via an application (ie a database).
+
```

Comment:
@j-griffith  once the PVC is successfully transferred,  are we expecting that, the source PVC object is wiped from API server for source namespace ?